metric/point.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"time"
+
+	"go.opencensus.io/exemplar"
+)
+
+// Point is a single data point of a time series.
+type Point struct {
+	// Time is the point in time that this point represents in a time series.
+	Time time.Time
+	// ValueType is the type of value held in this point.
+	ValueType ValueType
+	// Value is the value of this point. Prefer using ReadValue to switching on
+	// the value type, since new value types might be added.
+	Value interface{}
+}
+
+//go:generate stringer -type ValueType
+
+// ValueType is the type of value held in a point.
+type ValueType int
+
+// Value types. New value types may be added.
+const (
+	ValueTypeDouble ValueType = iota
```

Comment:
There is no doubles in Go, if the name is dictated by the spec, it should be flex-ed.

Reply:
I was on the fence about this. I agree it looks very strange. Will change it.